Sources/RingPublishingTracking/Public/RingPublishingTracking+Paid.swift

```diff
@@ -0,0 +1,151 @@
+        let event = eventsFactory.createShowOfferEvent(contentMetadata: contentMetadata,
+                                                       offerData: offerData,
+                                                       offerContextData: offerContextData,
+                                                       targetPromotionCampaignCode: targetPromotionCampaignCode)
+        reportEvent(event)
+    }
+
+    /// Reports showing offer teaser event
+    /// There is no possibility to start purchasing process flow from this place
+    ///
+    /// - Parameters:
+    ///   - contentMetadata: Content metadata
+    ///   - offerData: Data regarding the supplier of sales offers and offers themselves
+    ///   - offerContextData: Data regarding the offer context / content
+    ///   - targetPromotionCampaignCode: Offer id of given promotion / campaign
+    func reportShowOfferTeaserEvent(contentMetadata: ContentMetadata?,
+                                    offerData: OfferData,
+                                    offerContextData: OfferContextData,
+                                    targetPromotionCampaignCode: String?) {
+        let event = eventsFactory.createShowOfferTeaserEvent(contentMetadata: contentMetadata,
+                                                             offerData: offerData,
+                                                             offerContextData: offerContextData,
+                                                             targetPromotionCampaignCode: targetPromotionCampaignCode)
+        reportEvent(event)
+    }
+
+    /// Reports event of clicking button used to start purchasing process flow
+    ///
+    /// - Parameters:
+    ///   - contentMetadata: Content metadata
+    ///   - offerData: Data regarding the supplier of sales offers and offers themselves
+    ///   - offerContextData: Data regarding the offer context / content
+    ///   - termId: Id of specific purchase term / offer selected by user
+    ///   - targetPromotionCampaignCode: Offer id of given promotion / campaign
+    func reportPurchaseClickButtonEvent(contentMetadata: ContentMetadata?,
+                                        offerData: OfferData,
+                                        offerContextData: OfferContextData,
+                                        termId: String,
+                                        targetPromotionCampaignCode: String?) {
+        let event = eventsFactory.createPurchaseClickButtonEvent(contentMetadata: contentMetadata,
+                                                                 offerData: offerData,
+                                                                 offerContextData: offerContextData,
+                                                                 termId: termId,
+                                                                 targetPromotionCampaignCode: targetPromotionCampaignCode)
+        reportEvent(event)
+    }
+
+    // swiftlint:disable function_parameter_count
```

Comment:
to jest zaszłość z lokalnego testowania, czy ma zostać?

Reply:
nie, to wyłączenie lintera na ten przypadek, moze zostac

Reply:
Lint ma ustawioną regułę na maxymalną liczę parametrów 5 (w tym projekcie), a chciałem aby Android i iOS miały taki sam interface. No i trzeba zignorować tą regułę.